src/implicit.rs

```diff
@@ -137,19 +139,140 @@ pub fn resolve_weak(sequence: &IsolatingRunSequence, processing_classes: &mut [B
 ///
 /// <http://www.unicode.org/reports/tr9/#Resolving_Neutral_Types>
 #[cfg_attr(feature = "flame_it", flamer::flame)]
-pub fn resolve_neutral(
+pub fn resolve_neutral<D: BidiDataSource>(
+    text: &str,
+    data_source: &D,
     sequence: &IsolatingRunSequence,
     levels: &[Level],
+    original_classes: &[BidiClass],
     processing_classes: &mut [BidiClass],
 ) {
+    // e = embedding direction
     let e: BidiClass = levels[sequence.runs[0].start].bidi_class();
+    let e_is_l = e == BidiClass::L;
+    // N0. Process bracket pairs.
+
+    // Identify the bracket pairs in the current isolating run sequence according to BD16.
+    let bracket_pairs = identify_bracket_pairs(text, data_source, sequence, original_classes);
+
+    // For each bracket-pair element in the list of pairs of text positions
```

Comment:
I find the comments here a little hard to read, since they are verbatim citations from the UBA interspersed with the implementer’s actual comments; maybe we could mark the UBA citations as such with `// > ` or something similar?

Reply:
good call!